app/src/main/java/com/celzero/bravedns/service/PersistentState.kt

```diff
@@ -67,58 +69,47 @@ class PersistentState(private val context: Context) : SimpleKrate(context), Koin
     var tempBlocklistDownloadTime by longPref("temp_time_during_download", 0)
     var httpProxyPort by intPref("http_proxy_port", INVALID_PORT)
     var httpProxyHostAddress by stringPref("http_proxy_ipaddress", "")
-
     var killAppOnFirewall by booleanPref("kill_app_on_firewall", true)
     var allowByPass by booleanPref("allow_bypass", true)
     var allowDNSTraffic by booleanPref("dns_all_traffic", true)
     var dnsType by intPref("dns_type", 1)
     var prefAutoStartBootUp by booleanPref("auto_start_on_boot", true)
     var screenState by booleanPref("screen_state", false)
-
     var oldNumberRequests by intPref("number_request", 0)
     var oldBlockedRequests by intPref("blocked_request", 0)
     var numberOfRequests by longPref("dns_number_request", 0)
     var numberOfBlockedRequests by longPref("dns_blocked_request", 0)
-
     var backgroundEnabled by booleanPref("background_mode", false)
     var checkForAppUpdate by booleanPref("check_for_app_update", true)
-    private var connectedDnsName by stringPref("connected_dns_name", context.getString(R.string.dns_mode_3))
+    var connectedDnsName by stringPref("connected_dns_name", context.getString(R.string.dns_mode_3))
     var theme by intPref("app_theme", 0)
-    var notificationAction by intPref("notification_action", 1)
+    var notificationActionType by intPref("notification_action", 1)
     var isAddAllNetworks by booleanPref("add_all_networks_to_vpn", false)
-
     var lastAppRefreshTime by longPref("last_app_refresh_time", INIT_TIME_MS)
-
     var proxyType by stringPref("proxy_proxytype", AppMode.ProxyType.NONE.name)
     var proxyProvider by stringPref("proxy_proxyprovider", AppMode.ProxyProvider.NONE.name)
-
     private var _dnsCryptRelayCount by intPref("dnscrypt_relay", 0)
-
-    var orbotConnectionStatus: MutableLiveData<Boolean> = MutableLiveData()
-
+    var lastAppExitInfoTimestamp by longPref("prev_trace_timestamp", INIT_TIME_MS)
     var downloadIds by stringSetPref("download_ids", emptySet())
-
     var fetchFavIcon by booleanPref("fav_icon_enabled",
                                     BuildConfig.FLAVOR != Constants.FLAVOR_FDROID)
+    var appState by intPref("app_state", AppMode.AppState.ACTIVE.state)
+    var showWhatsNewChip by booleanPref("show_whats_new_chip", true)
 
+    var orbotConnectionStatus: MutableLiveData<Boolean> = MutableLiveData()
```

Comment:
Let's add one line comment against each of these prefs.

```

    // *** drop _?
    // when vpn is started by the user, this is set to true; set to false when user stops
    // the vpn. In case vpn crashes, this value remains true, which is expected.
    private var _vpnEnabled by booleanPref("enabled", false)
    
    // OOBE (out-of-the-box experience) screens are shown if the user
    // launches the app for the very first time (after install or post clear-data)
    var firstTimeLaunch by booleanPref("is_first_time_launch", true)
    
    // One among AppMode.BraveMode enum; 2's default, which is BraveMode.DNS_FIREWALL
    var braveMode by intPref("brave_mode", 2)
    
    // ...
    var firewallMode by intPref("firewall_mode", 1)
    
    // enable / disable logging dns and tcp/udp connections to db
    var logsEnabled by booleanPref("local_logs", true)
    
    // the last-set app version; useful to detect an update where the current
    // app version is going to be greater than the one stored, and so, update flow
    // can be triggered accordingly, if any,
    var appVersion by intPref("app_version", 0)
    
    // Last known time when app update checks were done (successful?)
    var lastAppUpdateCheck by longPref("app_update_last_check", 0)
    
    // total blocklists set by the user for RethinkDNS+ (server-side dns blocking)
    private var numberOfRemoteBlocklists by intPref("remote_block_list_count", 0)
    
    // total blocklists set by the user (on-device dns blocking)
    var numberOfLocalBlocklists by intPref("local_block_list_count", 0)
    
    // whether all udp connection except dns must be dropped
    var udpBlockedSettings by booleanPref("block_udp_traffic_other_than_dns", false)
    
    // ...
    var insertionCompleted by booleanPref("initial_insert_servers_complete", false)
    
    // user chosen blocklists stored custom dictionary indexed in base64
    var localBlocklistStamp by stringPref("local_block_list_stamp", "")
    
    // whether to drop packets when the source app originating the reqs couldn't be determined
    var blockUnknownConnections by booleanPref("block_unknown_connections", false)
    
    // whether the local blocklist files have been downloaded successfully
    // *** This is a redundant setting, get rid of it. The way to detect download is to 
    // *** check for timestamp and then presence of files within that timestamp-dir 
    var blocklistFilesDownloaded by booleanPref("download_block_list_files", false)
    
    // whether user has enable on-device blocklists
    var blocklistEnabled by booleanPref("enable_local_list", false)
   
    // ...
    var remoteBlocklistDownloadTime by longPref("remote_block_list_downloaded_time", 0)
    
    // the version (which is a unix timestamp) of the current on-device blocklist files
    var blocklistDownloadTime by longPref("local_block_list_downloaded_time", 0)
    
    // *** I rem asking to remove this; it is a code-smell. Use worker input / ouput
    // *** parameters instead to pass on the current timestamp from one worker function
    // *** to another
    // https://developer.android.com/topic/libraries/architecture/workmanager/advanced#params
    var tempBlocklistDownloadTime by longPref("temp_time_during_download", 0)
    
    // user set http proxy port
    var httpProxyPort by intPref("http_proxy_port", INVALID_PORT)
    
    // user set http proxy ip / hostname
    var httpProxyHostAddress by stringPref("http_proxy_ipaddress", "")
    
    // whether RethinkDNS should signal activity-manager to kill a firewalled app
    var killAppOnFirewall by booleanPref("kill_app_on_firewall", true)
    
    // whether apps subject to the RethinkDNS VPN tunnel can bypass the tunnel on-demand 
    // *** rename, allowBypass
    var allowByPass by booleanPref("allow_bypass", true)
    
    // trap all packets on port 53 to be sent to a dns endpoint or just the packets sent to vpn's dns-ip
    var allowDNSTraffic by booleanPref("dns_all_traffic", true)
    
    // user set among AppMode.DnsType enum; 1's the default which is DoH
    var dnsType by intPref("dns_type", 1)
    
    // whether the app must attempt to startup on reboot if it was running before shutdown
    var prefAutoStartBootUp by booleanPref("auto_start_on_boot", true)
    
    // *** rename, blockWhenDeviceLocked
    // user set preference whether firewall should block all connections when device is locked 
    var screenState by booleanPref("screen_state", false)
    
    // ...
    var oldNumberRequests by intPref("number_request", 0)
    
    // ...
    var oldBlockedRequests by intPref("blocked_request", 0)
    
    // total dns requests the app has served since installation (or post clear data)
    var numberOfRequests by longPref("dns_number_request", 0)
    
    // total dns requests blocked since installation
    var numberOfBlockedRequests by longPref("dns_blocked_request", 0)
    
    // *** rename, blockAppWhenBackground
    // whether to block connections from apps not in the foreground
    var backgroundEnabled by booleanPref("background_mode", false)
    
    // whether to check for app updates once-a-week (on website / play-store builds)
    var checkForAppUpdate by booleanPref("check_for_app_update", true)
    
    // ...
    var connectedDnsName by stringPref("connected_dns_name", context.getString(R.string.dns_mode_3))
    
    // the current light/dark theme; 0's the default which is "Set by System"
    var theme by intPref("app_theme", 0)
    
    // ...
    var notificationActionType by intPref("notification_action", 1)
    
    // *** rename, useMultipleNetworks
    // add all networks (say, both wifi / mobile) with internet capability to the vpn tunnel
    var isAddAllNetworks by booleanPref("add_all_networks_to_vpn", false)
    
    // ...
    var lastAppRefreshTime by longPref("last_app_refresh_time", INIT_TIME_MS)
    
    // ...
    var proxyType by stringPref("proxy_proxytype", AppMode.ProxyType.NONE.name)
    
    // ...
    var proxyProvider by stringPref("proxy_proxyprovider", AppMode.ProxyProvider.NONE.name)
    
    // *** rename, drop the _
    // total dnscrypt server currently connected to
    private var _dnsCryptRelayCount by intPref("dnscrypt_relay", 0)
    
    // ...
    var lastAppExitInfoTimestamp by longPref("prev_trace_timestamp", INIT_TIME_MS)
    
    // *** should use WorkManager' input / output params instead
    // ...
    var downloadIds by stringSetPref("download_ids", emptySet())
    
    // *** replace with isFDroidFlavour()
    // fetch fav icons for domains in dns request
    var fetchFavIcon by booleanPref("fav_icon_enabled", ...)
    
    // *** this should be eliminated
    var appState by intPref("app_state", AppMode.AppState.ACTIVE.state)
    
    // whether to show "what's new" chip on the homescreen, usually
    // shown after a update and until the user dismisses it
    var showWhatsNewChip by booleanPref("show_whats_new_chip", true)

```

Reply:
done, all of them...